### PR TITLE
21Calendar

### PR DESCRIPTION
**Pull request by timohahaaa**
https://timohahaaa.github.io/layout_calendar/

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -7,6 +7,38 @@
     <link rel="stylesheet" href="./styles/main.scss">
   </head>
   <body>
-    <h1>Calendar</h1>
+    <div class="calendar">
```
Comment (agniya-i) [inline src/index.html]:
```suggestion
    <div class="calendar calendar--start-day--wed calendar--month-length--30">
```

You should specify modifiers in this way

File: src/styles/main.scss [hunk 2]
```diff
@@ -1,3 +1,62 @@
...
+  .calendar__days:nth-child(#{$i}) {
+    &::before {
+      content: "#{$i}";
+      font-family: Arial, Helvetica, sans-serif;
+      font-size: 30px;
+    }
+  }
+}
+
+$days: mon 1, tue 2, wed 3, thu 4, fri 5, sat 6, sun 7;
```
Comment (agniya-i) [inline src/styles/main.scss]: It would be nice to create separate file for variables

File: src/styles/main.scss [hunk 3]
```diff
@@ -1,3 +1,62 @@
...
+  &__days {
+    position: relative;
+    box-sizing: border-box;
+    height: 100px;
+    width: 100px;
+    background-color: #eee;
+    border: 1px solid black;
+
+    cursor: pointer;
+    transition: 500ms;
```
Comment (agniya-i) [inline src/styles/main.scss]:
```suggestion
    transition: background-color 500ms, ...;
```
Specify transition properties

File: src/styles/main.scss [hunk 4]
```diff
@@ -1,3 +1,62 @@
...
+  }
+
+  &__days:hover {
+    background-color: pink;
+    transform: translateY(-20px);
+  }
```
Comment (agniya-i) [inline src/styles/main.scss]:
```suggestion
  &__days {
    position: relative;
    box-sizing: border-box;
    height: 100px;
    width: 100px;
    background-color: #eee;
    border: 1px solid black;

    cursor: pointer;
    transition: 500ms;
    
    &:hover {
      background-color: pink;
      transform: translateY(-20px);
     }
  }
```

Use SASS nesting

File: src/styles/main.scss [hunk 5]
```diff
@@ -1,3 +1,62 @@
-body {
-  margin: 0;
+.calendar {
```
Comment (agniya-i) [inline src/styles/main.scss:3]: Display a calendar in the middle of the screen

**Review by agniya-i — CHANGES_REQUESTED** [review]

File: src/styles/main.scss [hunk 6]
```diff
@@ -1,3 +1,70 @@
...
+  width: 707px;
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  padding: 10px;
+
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
```
Comment (maxim2310) [inline src/styles/main.scss]: its not a mistake, but we have a more easier way to center block like margin: 0 auto

File: src/styles/main.scss [hunk 7]
```diff
@@ -1,3 +1,70 @@
...
+  &--start-day--#{$day} > &__days:first-child {
+    margin-left: (($ceil-width + $gap-size) * $n);
+  }
+}
+
+@for $length from 28 through 31 {
+  &__days-#{$length} > div:nth-child(n + #{$length} + 1) {
+    display: none;
+  }
 }
```
Comment (maxim2310) [inline src/styles/main.scss]: this code is not working now, when calendar have &__month-length--30 the last desplayed day must be 30, not 31, and when calendar--start-day--wed left margin must be like calendar start from 3rd ceil
image

File: src/styles/utils/variables.scss [hunk 8]
```diff
@@ -0,0 +1,4 @@
+$days: mon 1, tue 2, wed 3, thu 4, fri 5, sat 6, sun 7;
+$n: 1, 2, 3, 4, 5, 6, 7;
```
Comment (maxim2310) [inline src/styles/utils/variables.scss]: you dont need this variable
```suggestion
```

**Review by maxim2310 — CHANGES_REQUESTED** [review]
Good job!!!
but make a sure that your styles have been applied

File: src/styles/main.scss (hunk 6 above)
Comment (timohahaaa) [reply src/styles/main.scss]: Yes, but then the block would place in the horisontal center but not vertical

File: src/styles/main.scss (hunk 7 above)
Comment (timohahaaa) [reply src/styles/main.scss]: @maxim2310 what should I do here?

File: src/styles/main.scss [hunk 9]
```diff
@@ -1,3 +1,71 @@
...
+    margin-left: (($ceil-width + $gap-size) * $n);
+  }
+}
+
+@for $length from 28 through 31 {
+  .calendar--month-length--#{$length}
+  > .calendar__days:nth-child(n + #{$length + 1}) {
+    display: none;
+  }
 }
```
Comment (maxim2310) [inline src/styles/main.scss]: as i see it already working, but would be better to use nesting of SCSS and move these func to .calendar or .calendar__day

**Review by maxim2310 — CHANGES_REQUESTED** [review]
Great work, the last push left)

File: src/styles/main.scss [hunk 10]
```diff
@@ -1,3 +1,71 @@
...
+@each $day, $n in $days {
+  .calendar--start-day--#{$day} > .calendar__days:first-child {
+    margin-left: (($ceil-width + $gap-size) * $n);
+  }
+}
+
+@for $length from 28 through 31 {
+  .calendar--month-length--#{$length}
+  > .calendar__days:nth-child(n + #{$length + 1}) {
+    display: none;
```
Comment (lerastarynets) [inline src/styles/main.scss]: Pls make use of SASS nesting here
for example
```
@each $day, $n in $days {
    &--start-day--#{$day} :first-child {
      margin-left: (($ceil-width + $gap-size) * $n);
    }
  }
```
and so on

**Review by lerastarynets — CHANGES_REQUESTED** [review]
Pls fix prev comments

**Review by agniya-i — CHANGES_REQUESTED** [review]
Good job! Let's fix several bugs

1. The month should start from Wednesday in this case, but it starts from Thursday (same for other modifier values)
Screenshot 2023-08-01 at 17 49 29

2. Fix styles for this case 
Screenshot 2023-08-01 at 17 53 55

File: src/styles/main.scss [hunk 11]
```diff
@@ -1,3 +1,62 @@
...
+
+  @for $i from 1 through 31 {
+    .calendar__days:nth-child(#{$i}) {
+      &::before {
+        content: "#{$i}";
+        font-family: Arial, Helvetica, sans-serif;
+        font-size: 30px;
+      }
+    }
+  }
```
Comment (maxim2310) [inline src/styles/main.scss]:
```suggestion
  @for $i from 1 through 31 {
    &__days:nth-child(#{$i}) {
      &::before {
        content: "#{$i}";
        font-family: Arial, Helvetica, sans-serif;
        font-size: 30px;
      }
    }
  }
```

File: src/styles/main.scss [hunk 12]
```diff
@@ -1,3 +1,62 @@
...
+
+  @each $day, $n in $days {
+    &--start-day--#{$day} > .calendar__days:first-child {
+      margin-left: (($ceil-width + $gap-size) * $n - ($ceil-width + $gap-size));
+    }
+  }
```
Comment (maxim2310) [inline src/styles/main.scss]:
```suggestion
  @each $day, $n in $days {
    &--start-day--#{$day} > &__days:first-child {
      margin-left: (($ceil-width + $gap-size) * $n - ($ceil-width + $gap-size));
    }
  }
```

File: src/styles/main.scss [hunk 13]
```diff
@@ -1,3 +1,62 @@
...
+
+  @for $length from 28 through 31 {
+    &--month-length--#{$length}
+    > .calendar__days:nth-child(n + #{$length + 1}) {
+      display: none;
+    }
+  }
```
Comment (maxim2310) [inline src/styles/main.scss:61]:
```suggestion
  @for $length from 28 through 31 {
    &--month-length--#{$length}
    > &__days:nth-child(n + #{$length + 1}) {
      display: none;
    }
  }
```

**Review by maxim2310 — CHANGES_REQUESTED** [review]
Almost on target)

**Review by polosanya — APPROVED** [review]
Nice!

**Pull request closed without merging**